### PR TITLE
Applied CS fixer and added config

### PR DESCRIPTION
**Pull request by dantleech**
This PR introduces a `.php_cs` file which contains the config for the `phpcsfixer` application which automatically fixes coding standards, and which we can adjust to our requirements.

The configuration is based on the `symfony` coding standard with the following additions:

````php
return Symfony\CS\Config\Config::create()
    ->fixers(array(
        'symfony',
        'concat_with_spaces',
        'ordered_use',
        '-phpdoc_indent',
        '-phpdoc_params',
    );
````

- Force spaces with string concatenation
- Order use statements alphabetically
- Do not indent phpdocs or phpdoc parameters
- Fixes the header

Additionally the default Symfony configuration:

- Removes unused `use` statements
- Adds fullstops `.` at the end of all documentation paragraphs.
- Lots more ..

### CODE REVIEW

**Comment by wachterjohannes** [comment]
+1

File: src/Sulu/Bundle/AdminBundle/Admin/AdminPool.php [hunk 1]
```diff
@@ -52,7 +55,7 @@ public function getNavigation()
         /** @var Navigation $navigation */
         $navigation = null;
         foreach ($this->pool as $admin) {
-            /** @var Admin $admin */
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Admin/AdminPool.php]: that is not good (= intellij needs this double star to know that is phpdoc-information

File: .php_cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+$header = <<<EOF
+This file is part of the Sulu CMS.
```
Comment (chirimoya) [inline .php_cs]: We should remove the CMS and only use 'Sulu'.

File: src/Sulu/Bundle/AdminBundle/Controller/ContentNavigationController.php [hunk 3]
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Controller/ContentNavigationController.php]: we never added this empty line at the start!

**Comment by dantleech** [comment]
Updated, fixed headers and do not remove the empty line after the opening tag.

**Comment by chirimoya** [comment]
ping ... ready?

**Comment by wachterjohannes** [comment]
@dantleech i have added this to my thesis project and we could also add this to the pre_commit hook locally to run it automatically! also we could add it to travis and it should fail if something change ... so we could detect if there are code style issues!

```bash
#!/bin/sh
#
#  .git/hooks/pre-commit
#

php-cs-fixer fix src
git add .
```

**Comment by dantleech** [comment]
@wachterjohannes it would be nice in theory, but in reality it takes something like 10 minutes to run on the Sulu code base. I think it would be better to run it once before every release and in the future maybe have it handled by continuous integration. /cc @chirimoya @danrot

**Comment by dantleech** [comment]
Adding it to travis would be nice, but only if we could somehow analyze only the code that has changed instead of everything.. should be possible.

**Comment by dantleech** [comment]
````
for file in `git diff --name-only origin/develop`; do phpcsfixer fix --dry-run $file; done
````

Will exist with `1` if any fixes were needed.

Will see about adding this to travis once this PR is merged.

**Comment by wachterjohannes** [comment]
@dantleech you only have to check the changed files in git => your third comment (= ...

perhaps we could run only over changed files? that should also be possible i think! perhaps i will test it today!

**Comment by wachterjohannes** [comment]
@dantleech that should work:

* first get changed files with: `git status --porcelain`
* combine them and pass it to php-cs-fixer (https://github.com/FriendsOfPHP/PHP-CS-Fixer#usage) 

there you can pass only files and you don't have to go over the full core!

**Comment by danrot** [comment]
+1 for applying on pre commit hook on only changed files. Then it shouldn't take too long... The only problem I can see is if somebody does a very big change, then the commit could take some time... We probably have to see in practice how long it actually takes.

**Comment by wachterjohannes** [comment]
@danrot i run it on the full codebase of my project, thats about 100 files, it takes about 30seconds. Thats good enough i think! because you shouldn't commit such a big change in one step!

**Comment by danrot** [comment]
IMHO waiting 30s for a commit is not acceptable in any case. It might happen that you do bigger refactorings from time to time. E.g. changing a method name using the refactoring tools in intellij can easily change a hundred files. A commit should remain a fast operation, otherwise developer will tend to do less commits.

**Comment by chirimoya** [comment]
Can be merged now?

**Pull request merged**